### PR TITLE
Initial migration from svotaw onnx-protobuf

### PR DESCRIPTION
**Pull request by svotaw**
Migration of code from svotaw/onnx-protobuf repo.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    name := "onnx-protobuf",
+  ): _*)
+
+ThisBuild / sonatypeProjectHosting := Some(
+  GitHubHosting("Microsoft", "SynapseML", "mmlspark-support@microsoft.com"))
+ThisBuild / homepage := Some(url("https://github.com/Microsoft/SynapseML"))
+ThisBuild / scmInfo := Some(
+  ScmInfo(
+    url("https://github.com/Microsoft/SynapseML"),
+    "scm:git@github.com:Azure/SynapseML.git"
```
Comment (memoryz) [inline build.sbt]: This needs to be updated right?

File: build.sbt [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  )
+)
+ThisBuild / developers := List(
+  Developer("mhamilton723", "Mark Hamilton",
+    "synapseml-support@microsoft.com", url("https://github.com/mhamilton723")),
+  Developer("svotaw", "Scott Votaw",
+    "synapseml-support@microsoft.com", url("https://github.com/svotaw"))
+)
+
+ThisBuild / licenses += ("MIT", url("https://github.com/Microsoft/SynapseML/blob/master/LICENSE"))
```
Comment (memoryz) [inline build.sbt]: This as well.

**Review by memoryz — APPROVED** [review]

**Comment by memoryz** [comment]
LGTM. Some additional thoughts on future work:
1. Can we submodule the onnx repo and code gen the OnnxMl file at build time? This way we'll always have the latest generated file if the proto file gets changed at source (not very often, but still gets updated every few months). Or is it not necessary?
2. Adding some basic unit test code to make sure the generated code can be loaded in Spark?

File: build.sbt [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+ThisBuild / scmInfo := Some(
+  ScmInfo(
+    url("https://github.com/Microsoft/SynapseML"),
+    "scm:git@github.com:Azure/SynapseML.git"
+  )
+)
+ThisBuild / developers := List(
+  Developer("mhamilton723", "Mark Hamilton",
+    "synapseml-support@microsoft.com", url("https://github.com/mhamilton723")),
+  Developer("svotaw", "Scott Votaw",
```
Comment (mhamilton723) [inline build.sbt:67]: Perhaps you should go first as you are the mastermind behind this :)

File: build.sbt (hunk 1 above)
Comment (svotaw) [reply build.sbt]: yeah, I'll update these, although for now I think we'll actually want to point to SynapseML for support.  I'm not sure I'll notice as quickly if someone does anything in this repo. But yeah, certainly want to remove refs to my private repo.

**Review by mhamilton723 — COMMENTED** [review]
Other than Jason's comments approve this!

**Comment by svotaw** [comment]
> LGTM. Some additional thoughts on future work:
> 
> 1. Can we submodule the onnx repo and code gen the OnnxMl file at build time? This way we'll always have the latest generated file if the proto file gets changed at source (not very often, but still gets updated every few months). Or is it not necessary?

It's a good idea about generation OnnxML.java at build time, and one of the things that we'd hope to do if this was moved over to the onnx team. It would be a nice addition. I imagine we could add an sbt command that grabbed the latest from the onnx repo and ran protoc on it.

As far as submodule, the shading stuff was finicky, and this is how I got it to work. I had hoped to add a submodule to our own SynapseML repo, but couldn't get it to work. It was much cleaner this isolated way, and easier to maintain.  Also, it seemed like this might be useful to someone else.

You'd know better than me whether any future changes would result in this not being backward compatible. :) We don't expose any of these APIs to our user, so I would hope there would be backward compatibility at the very least for just the APIs we use. I guess it's possible that someone uses a newer feature and it breaks something we when we try to split a model at a particular output.

> 2. Adding some basic unit test code to make sure the generated code can be loaded in Spark?

Yeah, certainly that could be done. For now, our own testing suite of SynapseML uses the public package generated from here and so we'd notice there if it stopped working. Given that there's no testing infra in this project or associated ADO resoures (for now), it would not be insignificant effort to add testing inside a Spark environment.

**Pull request merged**